diesel/src/sqlite/mod.rs

```diff
@@ -24,12 +24,19 @@ pub trait SqliteAggregateFunction<Args>: Default {
     /// The result type of the SQLite aggregate function
     type Output;
 
-    /// The `step()` method is called once for every record of the query
+    /// The `step()` method is called once for every record of the query.
+    ///
+    /// This is called through a C FFI, as such panics do not propagate to the caller. Panics are
+    /// caught and cause a return with an error value. The implementation must still ensure that
```

Comment:
As explained above I think we do not need to have that actually here, because I see no way how the state could be reused here.